src/coreclr/vm/prestub.cpp

```diff
@@ -2261,6 +2267,16 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT, CallerGCMode callerGCMo
     if (IsUnboxingStub())
     {
         pStub = MakeUnboxingStubWorker(this);
+
+#if defined(FEATURE_PORTABLE_ENTRYPOINTS) 
```

Comment:
This should be moved down and replace `if (!GetOrCreatePrecode()->SetTargetInterlocked(pStub->GetEntryPoint()))` so that it applies to all paths the return `Stub*`.